include/deal.II/differentiation/ad/ad_drivers.h

```diff
@@ -0,0 +1,1583 @@
+    void
+    TapedDrivers<ADNumberType, ScalarType, T>::enable_taping(
+      const types::tape_index,
+      const bool,
+      const types::tape_buffer_sizes,
+      const types::tape_buffer_sizes,
+      const types::tape_buffer_sizes,
+      const types::tape_buffer_sizes)
+    {
+      AssertThrow(false, ExcRequiresADNumberSpecialization());
+    }
+
+
+    template <typename ADNumberType, typename ScalarType, typename T>
+    void
+    TapedDrivers<ADNumberType, ScalarType, T>::disable_taping(
+      const types::tape_index,
+      const bool)
+    {
+      AssertThrow(false, ExcRequiresADNumberSpecialization());
+    }
+
+
+    template <typename ADNumberType, typename ScalarType, typename T>
+    void
+    TapedDrivers<ADNumberType, ScalarType, T>::print_tape_stats(
+      std::ostream &,
+      const types::tape_index)
+    {
+      AssertThrow(false, ExcRequiresADNumberSpecialization());
+    }
+
+
+    template <typename ADNumberType, typename ScalarType, typename T>
+    ScalarType
+    TapedDrivers<ADNumberType, ScalarType, T>::value(
+      const types::tape_index,
+      const std::vector<ScalarType> &)
+    {
+      AssertThrow(false, ExcRequiresADNumberSpecialization());
+      return ScalarType(0.0);
+    }
+
+
+    template <typename ADNumberType, typename ScalarType, typename T>
+    void
+    TapedDrivers<ADNumberType, ScalarType, T>::gradient(
+      const types::tape_index,
+      const std::vector<ScalarType> &,
+      Vector<ScalarType> &)
+    {
+      AssertThrow(false, ExcRequiresADNumberSpecialization());
+    }
+
+
+    template <typename ADNumberType, typename ScalarType, typename T>
+    void
+    TapedDrivers<ADNumberType, ScalarType, T>::hessian(
+      const types::tape_index,
+      const std::vector<ScalarType> &,
+      FullMatrix<ScalarType> &)
+    {
+      AssertThrow(false, ExcRequiresADNumberSpecialization());
+    }
+
+
+    template <typename ADNumberType, typename ScalarType, typename T>
+    void
+    TapedDrivers<ADNumberType, ScalarType, T>::values(
+      const types::tape_index,
+      const unsigned int,
+      const std::vector<ScalarType> &,
+      Vector<ScalarType> &)
+    {
+      AssertThrow(false, ExcRequiresADNumberSpecialization());
+    }
+
+
+    template <typename ADNumberType, typename ScalarType, typename T>
+    void
+    TapedDrivers<ADNumberType, ScalarType, T>::jacobian(
+      const types::tape_index,
+      const unsigned int,
+      const std::vector<ScalarType> &,
+      FullMatrix<ScalarType> &)
+    {
+      AssertThrow(false, ExcRequiresADNumberSpecialization());
+    }
+
+
+#  ifdef DEAL_II_WITH_ADOLC
+
+    // Specialization for taped ADOL-C auto-differentiable numbers.
+    //
+    // Note: In the case of ADOL-C taped numbers, the associated scalar
+    // type is always expected to be a double. So we need to make a further
+    // specialization when ScalarType is a float.
+    template <typename ADNumberType>
```

Comment:
make this a doxygen-style comment so it shows up on the website.

you can copy the documentation of each of the functions using the `@copydoc` tag